src/mono/mono/mini/interp/transform.c

```diff
@@ -3655,10 +3662,12 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 			}
 #endif
 		}
-		td->last_ins->flags |= INTERP_INST_FLAG_CALL;
+		init_last_ins_call (td);
```

Comment:
we should set `call_args` here and not allocate the call info below, for non-call instructions

Reply:
Now, there is a check `if (td->last_ins->flags & INTERP_INST_FLAG_CALL) {` so I haven't set `call_args` here. Let me know if it should be updated.